modules/shared_vpc_access/main.tf

```diff
@@ -19,40 +19,49 @@ data "google_project" "service_project" {
 }
 
 locals {
-  gke_shared_vpc_enabled = contains(var.active_apis, "container.googleapis.com")
-  gke_s_account = local.gke_shared_vpc_enabled ? format(
-    "service-%s@container-engine-robot.iam.gserviceaccount.com",
-    data.google_project.service_project.number,
-  ) : ""
+  apis = {
+    "container.googleapis.com" : format("service-%s@container-engine-robot.iam.gserviceaccount.com", data.google_project.service_project.number),
+    "dataproc.googleapis.com" : format("service-%s@dataproc-accounts.iam.gserviceaccount.com", data.google_project.service_project.number),
+  }
+  gke_shared_vpc_enabled      = contains(var.active_apis, "container.googleapis.com")
```

Comment:
We shouldn't need these individual _enabled locals.

Reply:
This check will be needed for the `hostServiceAgentUser` binding but if that can be removed, we can get rid of this as well.

Reply:
Right but why do we need a conditional for dataproc? We should *only* use `gke_shared_vpc_enabled` for the hostServiceAgentUser part.

Reply:
Oops didnt see dataproc_enabled was still there, will clean up :)